### ISSUE TITLE
feat: allow to modify common labels of the resources

### ISSUE THREAD

**Issue by ikatlinsky** [apache/apisix-helm-chart#455]
**Issue**

Currently, common label structure and some values are hardcoded in the following form

```yaml
{{- define "apisix.selectorLabels" -}}
app.kubernetes.io/name: {{ include "apisix.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
``` 

There is no way to override `app.kubernetes.io/instance` at all, while we can do this for `app.kubernetes.io/name` using `nameOverride` property of the values file.

In our company we have a standard for labelling resourcing which is the following:

```yaml
{{- define "apisix.selectorLabels" -}}
app.kubernetes.io/name: {{ .Release.Name }}
app.kubernetes.io/instance: {{ include "apisix.name" . }}
app.kubernetes.io/component: {{ include "apisix.name" . }}
{{- end }}
``` 
That implies a hierarchy of labels: name -> instance -> component.

So, basically, we need to have 2 things:
1. Ability to override values for default labels
2. Ability to add own labels

**Proposed solution**

1. Modify `apisix.selectorLabels` template to allow override of all labels

```yaml
{{- define "apisix.selectorLabels" -}}
{{ if .Values.gateway.labelsOverride }}
{{ tpl (.Values.gateway.labelsOverride | toYaml) . }}
{{ else }}
app.kubernetes.io/name: {{ include "apisix.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{ end }}
{{- end }}
```

2. Provide overrides for gateway in values file

```yaml
gateway:
  ...
  labelsOverride:
    app.kubernetes.io/name: "{{ .Release.Name }}"
    app.kubernetes.io/instance: '{{ include "apisix.name" . }}'
    app.kubernetes.io/component: '{{ include "apisix.name" . }}'
```

**Comment by ikatlinsky** [comment]
Just want to know if the proposed option is ok and is not breaking anything. In case the solution does not match any best practices - I am open to proposals and improvements.

**Comment by tokers** [comment]
I think is feasible and reasonable.

**Comment by ikatlinsky** [comment]
@tokers ok, thanks for the response, please, assign ticket to me, will provide PR

**Comment by Gallardot** [comment]
#456 has been merged, I close this one, thanks
